.github/workflows/run-tests.yml

```
@@ -15,22 +14,15 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+  test:
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v4
-
       # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
-
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
+      - name: Build and Run tests
         run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
+          cargo build
+          cargo clippy
```

We usually deny warnings here to turn them to errors:

```suggestion
          cargo clippy -- -D warnings
```

Warnings in the CI are easily missable

---

done